conductr_cli/conduct_backup.py

```diff
@@ -0,0 +1,262 @@
+        log = logging.getLogger(__name__)
+        log.info("Processing bundle : " + bundle_core_info.bundle_id)
+
+        bundle_files_response = bundle_files(self.args, bundle_core_info.bundle_id)
+        bundle_validated = self.backup_bundle_file(backup_path, bundle_files_response[0], bundle_core_info)
+        bundle_conf_validated = False
+
+        # the config is also present
+        if len(bundle_files_response) == 2:
+            bundle_conf_validated = self.backup_bundle_conf(backup_path, bundle_files_response[1],
+                                                            bundle_core_info.configuration_digest)
+
+        validation_failed = bundle_validated is False or bundle_conf_validated is False
+        if validation_failed:
+            raise ConductBackUpError(str.format("Digest validations failed for {0}", self.bundle_id_or_name))
+
+    def backup_members(self, backup_path):
+        members_info = members(self.args)
+        members_json_path = os.path.join(backup_path, "members.json")
+        write_string_to_file(members_json_path, members_info)
+
+    def backup_agents(self, backup_path):
+        agents_info = agents(self.args)
+        agents_info_path = os.path.join(backup_path, "agents.json")
+        write_string_to_file(agents_info_path, agents_info)
+
+    @staticmethod
+    def remove_bundle(backup_path, bundle_info: BundleCoreInfo):
+        try:
+            bundle_file_path = str.format("{0}.zip", os.path.join(backup_path, bundle_info.bundle_name_with_digest))
+            bundle_conf_path = str.format("{0}.zip", os.path.join(backup_path, bundle_info.configuration_digest))
+            delete_file(bundle_file_path)
+            delete_file(bundle_conf_path)
+        except OSError:
+            error_message = str.format("Error rolling back Bundle {0} from backup.", bundle_info.bundle_id)
+            raise ConductBackUpError(error_message)
+
+    @staticmethod
+    def backup_bundle_json(backup_path, bundle_data):
+        bundles_json_path = os.path.join(backup_path, "bundles.json")
+        write_string_to_file(bundles_json_path, bundle_data)
+
+    @staticmethod
+    def backup_bundle_file(backup_path, bundle_file, bundle_info):
+        bundle_file_path = str.format("{0}.zip", os.path.join(backup_path, bundle_info.bundle_name_with_digest))
+        write_file(bundle_file_path, bundle_file)
+        return validate_artifact(bundle_file_path, bundle_info.bundle_digest)
+
+    @staticmethod
+    def backup_bundle_conf(backup_path, bundle_conf, configuration_digest):
+        bundle_conf_path = str.format("{0}.zip",
+                                      os.path.join(backup_path, configuration_digest))
+        write_file(bundle_conf_path, bundle_conf)
+        bundle_conf_validated = validate_artifact(bundle_conf_path, configuration_digest)
+        return bundle_conf_validated
+
+    @staticmethod
+    def compress_backup(output_path, backup_path, backup_name):
```

Comment:
@longshorej do we care about digests for the final zip ? I thought only bundles needed digests.